disnake/components.py

```diff
@@ -783,28 +839,401 @@ def to_dict(self) -> TextInputPayload:
         return payload
 
 
+class Section(Component):
+    """Represents a section from the Discord Bot UI Kit (v2).
+
+    This allows displaying an accessory (thumbnail or button) next to a block of text.
+
+    .. note::
+        The user constructible and usable type to create a
+        section is :class:`disnake.ui.Section`.
+
+    .. versionadded:: 2.11
+
+    Attributes
+    ----------
+    accessory: Union[:class:`Thumbnail`, :class:`Button`]
+        The accessory component displayed next to the section text.
+    components: List[:class:`TextDisplay`]
+        The text items in this section.
+    """
+
+    __slots__: Tuple[str, ...] = ("accessory", "components")
+
+    __repr_info__: ClassVar[Tuple[str, ...]] = __slots__
```

Comment:
Hmm, `Component` has `.__repr_info__`, but `.ui.UIComponent` has `.__repr_attributes__`. Is this desirable?
(I suggest we use the same class variable name everywhere)

Reply:
Hm, good point :thinking: 
I'd say `__repr_attributes__` sounds more accurate? `*_info` could mean anything, really. Both app commands and non-ui components use the latter, though we'd want to rename those in a separate PR to keep the scope somewhat concise in this PR